src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -333,15 +333,14 @@ private class FunctionActivityTracker : IDisposable
 
             private readonly IMetricsPublisher _metricsPublisher;
             private readonly ILinuxContainerActivityPublisher _linuxContainerActivityPublisher;
-            private readonly object _runningFunctionsSyncLock = new object();
             private readonly Timer _activityTimer;
             private readonly ILogger<MetricsEventManager> _logger;
 
             private ulong _totalExecutionCount = 0;
             private int _activeFunctionCount = 0;
             private int _functionActivityFlushInterval;
             private ConcurrentQueue<FunctionMetrics> _functionMetricsQueue = new ConcurrentQueue<FunctionMetrics>();
-            private List<FunctionStartedEvent> _runningFunctions = new List<FunctionStartedEvent>();
+            private ConcurrentDictionary<Guid, FunctionStartedEvent> _runningFunctions = new ConcurrentDictionary<Guid, FunctionStartedEvent>();
```

Comment:
Yep agreed - I dug into this more as well and tried to add detail in the PR description. The locking internals have changed since when those changes have made - given we're on .NET 5+ this is a much less contentious path and results in more throughput.

I'd love to get the anon/benchmark bits in so we can add Crank data to each of the PRs here - thoughts on https://github.com/Azure/azure-functions-host/pull/7941#issuecomment-986947273? It'd make it much easier to provide numbers for each, though they're all small and add-up I totally agree it's worthwhile.